falcon/routing/compiled.py

```diff
@@ -362,4 +358,4 @@ def conflicts_with(self, segment):
                 #   /foo/all
                 return True
```

Comment:
I'm not sure that's a good idea.  Personally I would rather have falcon raise an error when I do something stupid than have it try to work around it.  It's may be helpful when you first write it, but if you someone else comes back to debug that code, it's not clear if that path should be reachable or not and involves a deep dive into falcon to figure it out.